src/sharedHooks/useWatchPosition.ts

```diff
@@ -0,0 +1,125 @@
+import Geolocation, {
+  GeolocationError,
+  GeolocationResponse
+} from "@react-native-community/geolocation";
+import _ from "lodash";
+import { useEffect, useState } from "react";
+import {
+  RESULTS as PERMISSION_RESULTS
+} from "react-native-permissions";
+import {
+  checkLocationPermission,
+  shouldFetchObservationLocation,
+  TARGET_POSITIONAL_ACCURACY
+} from "sharedHelpers/shouldFetchObservationLocation.ts";
+import useStore from "stores/useStore";
+
+interface UserLocation {
+  latitude: number,
+  longitude: number,
+  positional_accuracy: number
+}
+
+interface WatchPosition {
+  hasLocation: boolean,
+  isFetchingLocation: boolean,
+  locationPermissionNeeded: boolean,
+  userLocation: UserLocation
+}
+
+const geolocationOptions = {
+  distanceFilter: 0,
+  enableHighAccuracy: true,
+  maximumAge: 0,
+  timeout: 2000
+};
+
+const useWatchPosition = ( options: {
+  retry: boolean
+} ): WatchPosition => {
+  const updateObservationKeys = useStore( state => state.updateObservationKeys );
```

Comment:
I strongly advise removing this and any observation-specific logic from this hook. It's called watchPosition, so it should just watch the position. If consumers of this module need to do other stuff, they should pass in callbacks as arguments.

In general, you want code to be modular (i.e. reusable) and encapsulated (with as few external dependencies as possible).  There are all kinds of reasons we might want to watch the user's location, so we don't want to tie this functionality to one specific use case.